--- gift-0.11.8.1/src/plugin.c.gcc4	2004-11-11 20:52:16.000000000 -0600
+++ gift-0.11.8.1/src/plugin.c	2005-06-27 13:56:32.000000000 -0500
@@ -128,11 +128,11 @@
 	size_t  msgwr = 0;                                                     \
 	va_list args;                                                          \
 	if (pfx1)                                                              \
-		msgwr += snprintf (APPENDMSG, "%s: ", STRING_NOTNULL(pfx1));       \
+		msgwr += snprintf (msg + msgwr, sizeof (msg) - msgwr - 1, "%s: ", STRING_NOTNULL(pfx1));       \
 	if (pfx2)                                                              \
-		msgwr += snprintf (APPENDMSG, "[%s]: ", STRING_NOTNULL(pfx2));     \
+		msgwr += snprintf (msg + msgwr, sizeof (msg) - msgwr - 1, "[%s]: ", STRING_NOTNULL(pfx2));     \
 	va_start (args, fmt);                                                  \
-	vsnprintf (APPENDMSG, fmt, args);                                      \
+	vsnprintf (msg + msgwr, sizeof (msg) - msgwr - 1, fmt, args);                                      \
 	va_end (args);
 
 static int wrapper_dbg (Protocol *p, char *fmt, ...)
